### ISSUE TITLE
Libreoffice 5: global appmenu stop working

### ISSUE THREAD

**Issue by pablitok** [lestcape/Gnome-Global-AppMenu#40]
Hi, I'm using ubuntu gnome 17.04 with gnome-shell 3.24.2 and LibreOffice 5.3.1.2 30m0(Build:2) and frequently the menu disappears after a couple times switching windows with alt+tab and have to close libreoffice or restart gnome shell with alt+f2;r to get the global menu working again

**Comment by lestcape** [comment]
Hi. Libreoffice have a big menu. It can cause troubles in the panel because could have not space to be displayed in the default configuration of gnome shell. I have ubuntu 16.04 and is not ubuntu gnome, also I have not gnome shell 3.24.2. I can not note this problem with my current configuration. Really a big change was introduced recently, maybe i have not the time to test all things...

The question is: Can you reproduced the problem when you want?
I need to know how something...
How you lost the menu?
You see the icon and the application name, but not the menu?
When you lost the global menu, libreoffice recover the menu?
The menu have the ability to hidden it if you preform a click inside the icon and the application name. Click there resolve the problem?

**Comment by pablitok** [comment]
I've updated the extension several times, but the problem persists 
(it's not something from the last release), Tonight I'll try to pin 
down the steps to reproduce the  problem and i'll write you

**Comment by HEXcube** [comment]
This issue can be reproduced in apps with long startup times, like Inkscape, GIMP, Atom, etc in my case. And sometimes Chrome too. However, you're less likely to encounter this error on fast machines. I don't see this often on my work machine whereas it's a pretty regular thing on my home notebook. As a workaround, I just restart shell (Alt+F2 and type r) while keeping the app open.

**Comment by lestcape** [comment]
@HEXcube can you described how the menu is lost?

- You see the icon and the application name, but not the menu?
- When you lost the global menu, the application recovered the menu?
- Swap between different windows and return to the offending window could resolve the problem?

**Comment by pablitok** [comment]
In my case:
* I lost evertything: icon, name and menu
* Libreoffice didn't recovered the menu
* The only way to recover the menu was restarting gnome-shell with 
alt+f2 r

**Comment by lestcape** [comment]
@pablitok if you tell me that you only lost the menu but not the icons, i can try to find a solution, also if word recover the menu i can try (worse but i can try). If you lost all... I really don't know what occurs...

**Comment by pablitok** [comment]
Give me a sec and i'll try to reproduce everything.

**Comment by lestcape** [comment]
Will be nice if you can reproduced it during the excution in the terminal of "journalctl -f"... This will output all things...

**Comment by pablitok** [comment]
So, I opened libre calc and everything was fine, then I opened a .ods 
with several pages, after that, alt+tab, then <super>, then alt+tab 
again and now the menu is gone, but the icon is still there.
If you wanna we can chat somewhere.
Sorry if my english is not good, is not my primary language

**Comment by lestcape** [comment]
@pablitok won't worried.... At less not for me... Maybe for other that are listen here... Take this like a chat if you want (for me not matter)... Also i don't speak English so, I need to say sorry too... 

What you say now is several time better, because i can try to find a solution for this... You are tell me that you see the menu and then it will gone after some alt+tab?

**Comment by pablitok** [comment]
Yeah, but not when i did alt+tab with an empty libreoffice calc 
document, only after I've loaded a "big" document (maybe 20 sheets)

**Comment by lestcape** [comment]
Ok, If the problem is inside the 20 sheets i really can not do anything, because that internal to libre office, so i will think that this just delay the export of the menu en off and for that reason it will not displayed, but well, I this can not explain why you lost a menu that you already have...

**Comment by HEXcube** [comment]
> @HEXcube can you described how the menu is lost?
> 
> You see the icon and the application name, but not the menu?
> When you lost the global menu, the application recovered the menu?
> Swap between different windows and return to the offending window could resolve the problem?

@lestcape  Sorry for the late reply, but I can't reproduce the issue now. :disappointed: I'm using the version from GNOME Extensions site now. Was there any bugfixes in recent updates? Else I'll just post the results here, when I encounter the issue again.

From what I remember, the Global Menu wouldn't be shown at all, not even the icon or name. It'll be just the GNOME AppMenu that's visible. However, pressing `Alt` would sometimes reveal the dropdown, but not the Global Menu itself. Search in GNOME Shell would work too.

I think Global Menu may not have been initialized properly on login, and that maybe why it get's fixed when I restart shell with `Alt+F2 and r`.

**Comment by HEXcube** [comment]
LibreOffice 5 works fine for me, though I don't use it heavily. However, there's a menu overflow problem, which I'm not sure is related to this issue:

![LibreOffice - Global Menu Overflow](https://user-images.githubusercontent.com/3755396/28246634-51beb6ea-6a3c-11e7-84d9-2f4026992915.png)

I'm using the latest stable LibreOffice 5.3 from fresh channel PPA:

![LibreOffice - About](https://user-images.githubusercontent.com/3755396/28246646-780beef8-6a3c-11e7-930e-d25b09d7122f.png)

**Comment by lestcape** [comment]
@HEXcube I will think is a problem in an specific version of gnome that i don't have, so i can not reproduced it... Probably the menu crash because of a different code in the gnome shell core or because a hack of some specific extension... I don't know... At less i don't try to fix anything, so if is fixed for you that’s fine, but i don't fixed it or try to fixed it...

About menu overflow problem, well yes there are, how can i fixed it? I really can not, because if you have not space to show the menu well, it will not be showed...  You also have now a way to control the wrap mode of the menu name inside the panel and do not let clutter to wrap the text and ellipsis it. This is set in the option Wrap the menu labels.  This will resolved the menu overflow problem, but then will ellipsis the labels of the menu to be adapted to the real available size, both behaviours are annoying, because the real problem is that there are not space.

You need to create the space in the panel using an extension to move the clock to the right... That is the only real solution. What it's in my hands it create a mechanism to move the clock inside the global menu, but really this is something that i don't want to do here, it's a gnome shell problem, not a global menu problem.

**Comment by HEXcube** [comment]
> About menu overflow problem, well yes there are, how can i fixed it? I really can not, because if you have not space to show the menu well, it will not be showed well...

@lestcape  I've discussed a potential solution in another issue here: https://github.com/lestcape/Gnome-Global-AppMenu/issues/20#issuecomment-315813942

**Comment by HEXcube** [comment]
Finally, I reproduced the issue on Atom. Happened on my work machine (GNOME Shell 3.20 running on Ubuntu 16.04LTS). The menu'd appear empty when Atom is loaded. Pressing Alt doesn't show the menu either (it used to show menu dropdown earlier). I've enabled Global Menu's options to hide appname and replace AppMenu. See:

![atom - no global menu](https://user-images.githubusercontent.com/3755396/28330567-ba3319b6-6c0b-11e7-8610-86526257271e.png)

Surprisingly, HUD and Menu search in Shell still worked:

![atom - hud](https://user-images.githubusercontent.com/3755396/28330616-e08b3dfa-6c0b-11e7-84af-8ed7af6e551f.png)

But, they both stopped working after the first use of HUD :worried: :

![atom - empty hud](https://user-images.githubusercontent.com/3755396/28330663-0cd13982-6c0c-11e7-8e59-5db0129df528.png)

I the end I just restarted Shell while keeping Atom open and then everything started working well.

Now to answer ur questions:

> You see the icon and the application name, but not the menu?

Yep

> When you lost the global menu, the application recovered the menu?

No, there were no menu at all

> Swap between different windows and return to the offending window could resolve the problem?

No, Global Menu'd be present on other apps but Atom still wouldn't have any menu when I return to it

**Comment by lestcape** [comment]
This is a difficult issue, because i can not reproduced it and apparently not always happen ... Thanks for the information... I think, it will take a long to be fixed... I need to see this of find a something that could be the cause.

**Comment by lestcape** [comment]
I fixed some related problem, can anyone confirmed that the problem is gone or continues?

**Comment by HEXcube** [comment]
Didn't get notified of the update, coz maintainer of extension on GNOME Online site ([Link](https://extensions.gnome.org/extension/1250/gnome-global-application-menu/)) isn't releasing updates. :disappointed: Anyway, I manually downloaded from GitHub and reinstalled.

LibreOffice seems to be working fine in all 3 modes (Global Menu, HUD and from GNOME Shell search). Atom's better than before, now shows menu without needing Shell restart. And works in GNOME Shell search. Global Menu and HUD does show items, but they don't work when clicked on. :open_mouth: 

![Atom after Global Menu update](https://user-images.githubusercontent.com/3755396/28744215-b26c1496-7479-11e7-910a-6470ef2208dc.png)

This seems to be an Atom specific issue. With migration to GTK3 by Electron (Atom's base), it'll use native menus and let's hope it solves the bug.

**Comment by HEXcube** [comment]
With LibreOffice working fine this particular issue seems to be solved already. Lots of duplicates for this bug report maybe coz users are still on older versions from the [GNOME Extensions site](https://extensions.gnome.org/extension/1250/gnome-global-application-menu/) or they simply didn't update from GitHub.

**Comment by lestcape** [comment]
@HEXcube https://github.com/lestcape/Gnome-Global-AppMenu/issues/52
